docs/developing_iree/ccache.md

```diff
@@ -0,0 +1,55 @@
+# Using `ccache` to build IREE
+
+[`ccache`](https://ccache.dev/) is a compilation cache. In principle, just prepending compiler invocations with `ccache` is all one needs to enable it, e.g.
+```shell
+ccache clang foo.c -c -o foo.o
+```
+takes care of executing `clang` with these arguments and caches the output file `foo.o`. The next invocation then skips executing `clang` altogether.
```

Comment:
Could expand on this introduction:

* typical build time savings on one machine
* comparison vs using CMake on its own

(Personally I have fast enough builds on a threadripper without using `ccache` or Bazel RBE)

Reply:
done!